SJChu: Code Updating for PR (dependency issues)

chuchu275 opened the pull request:
**1. colmap_colab.ipynb**
- COLMAP 설치 과정에서 FLANN 의존성 문제 해결을 위해 libflann-dev 설치 과정을 추가하였습니다.

**2. nerf_wandb_colab.ipynb**
- Colab 업데이트로 인한 기본 파이썬 버전 변경에 따른 TF 1.15 버전 설치 불가 문제로, CUDA 및 cudnn, 파이썬, TF를 모두 재설치하는 코드를 추가하였습니다.
- CUDA==10.0, cudnn==7.6, python==3.7, protobuf==3.20.1, tensorflow==1.15 버전으로 설치하였습니다.
- Mesh 코드를 실행하기 위한 config.txt 생성 셀에서의 make_config("dataset_type", 'llff') 코드는 실험 환경에 따라 상이함을 확인하여, 코드 포함 유무를 재확인해야 할듯 합니다.
- config.txt 파일에 맞추어 변수명을 변경하고, 다시 GPU를 설정하는 셀을 추가하였습니다.

**3. extract_mesh_colab.ipynb**
- 의존 패키지 설치에 따른 !apt-get update를 추가하였습니다.
- imagemagick을 확실히 설치하는 코드를 추가하였습니다.
- 이전 단계에서 TensorFlow 버전 변경으로 인해 호환성을 보장하기 위해 다시 임포트하는 코드를 추가하였습니다.

chuchu275 committed:
Code Updating for PR (dependency issues)

1. colmap_colab.ipynb
- COLMAP 설치 과정에서 FLANN 의존성 문제 해결을 위해 libflann-dev 설치 과정을 추가하였습니다.

2. nerf_wandb_colab.ipynb
- Colab 업데이트로 인한 기본 파이썬 버전 변경에 따른 TF 1.15 버전 설치 불가 문제로, CUDA 및 cudnn, 파이썬, TF를 모두 재설치하는 코드를 추가하였습니다.
- CUDA==10.0, cudnn==7.6, python==3.7, protobuf==3.20.1, tensorflow==1.15 버전으로 설치하였습니다.
- Mesh 코드를 실행하기 위한 config.txt 생성 셀에서의 make_config("dataset_type", 'llff') 코드는 실험 환경에 따라 상이함을 확인하여, 코드 포함 유무를 재확인해야 할듯 합니다.
- config.txt 파일에 맞추어 변수명을 변경하고, 다시 GPU를 설정하는 셀을 추가하였습니다.

3. extract_mesh_colab.ipynb
- 의존 패키지 설치에 따른 !apt-get update를 추가하였습니다.
- imagemagick을 확실히 설치하는 코드를 추가하였습니다.
- 이전 단계에서 TensorFlow 버전 변경으로 인해 호환성을 보장하기 위해 다시 임포트하는 코드를 추가하였습니다.

ProtossDragoon commented on notebooks/colmap_colab.ipynb:
```diff
@@ -1,268 +1,266 @@
+      "outputs": [],
+      "source": [
+        "!sudo apt-get install libmetis-dev # https://github.com/colmap/colmap/issues/1469\n",
+        "!sudo apt-get install libflann-dev # Required for approximate nearest neighbor searches in COLMAP"
```
이 부분이 추가되었다는 말씀이시죠? 저희가 COLMAP 버전 고정을 하지 않았더니 새로운 버전의 COLMAP 에서 의존 패키지가 추가되는 등 괜히 복잡한 문제가 생겼나봅니다.

ProtossDragoon replied:
제가 돌려보고 COLMAP 버전을 찾아 고정하려고 했는데, 혹시 PR 코멘트 확인하시면서 여유가 되신다면 - 지금처럼 git tag 를 검색하는 방식 대신, 릴리즈 태그를 고정하는 방식으로 변경해 주시거나 여유가 없다면 사용된 COLMAP 버전만 주석으로 덧붙여 주시면 감사하겠습니다.

ProtossDragoon replied:
제가 돌려보고 COLMAP 버전을 찾아 고정하려고 했는데 제 코랩 계정 이슈가 있네요.

혹시 PR 코멘트 확인하시면서 여유되신다면 - 현재같이 적절한 COLMAP 버전을 찾기 위해 git tag 를 검색하는 방식 대신 릴리즈 태그를 고정하는 방식으로 변경해 주실 수 있을까요? 여유가 없다면 사용된 COLMAP 버전만 주석으로 덧붙여 주셔도 감사하겠습니다.

chuchu275 replied:
안녕하세요, 저희가 다음주까지 대입 원서 접수가 마무리되어야 해서 실험을 병행한다고 부득이 늦게 답 드려 정말 죄송합니다.

PR #29에서 COLMAP 버전을 최신 버전으로 바꿔주셔서, 거기에 맞추어 다른 의존성도 해결하였습니다.
확인해보니 버전은
COLMAP 3.11.0.dev0 -- Structure-from-Motion and Multi-View Stereo
(Commit 689e85ba)
이라고 합니다.

!git clone https://github.com/colmap/colmap.git
%cd colmap

!git checkout 689e85ba
로 변경해보시는 것도 괜찮을 것 같습니다!

ProtossDragoon commented on notebooks/nerf_wandb_colab.ipynb:
```diff
@@ -55,13 +55,20 @@
       },
       "source": [
-        "### TF 런타임 변경"
+        "### CUDA 10.0 설치"
```
아아... 고생 많으셨습니다.

yyongjae replied:
오우 쿠다 버전을 바꾸셨나보네요,, 투자한 시간이 느껴집니다 고생하셨어요..

chuchu275 replied:
T4 GPU 기준으로 설정하였습니다!

ProtossDragoon commented on notebooks/nerf_wandb_colab.ipynb:
```diff
@@ -306,6 +397,53 @@
+        "import os\n",
+        "os.environ[\"CUDA_VISIBLE_DEVICES\"] = \"0\""
+      ]
+    },
```
실험 환경에 따라 상이하다고 말씀주셨는데, COLAB 이외의 환경을 말씀주시는 것일까요?

chuchu275 replied:
"make_config(\"dataset_type\", 'llff') # If the execution fails, remove the below code and try running the code again.\n",
부분을 말씀해주시는 거라면,
저희가 명확하지 않아 환경에 따라 상이하다고 말씀드렸는데
다시 실험해보니, COLAB 환경에서는 삭제해도 무관할 것 같습니다!

ProtossDragoon commented on notebooks/extract_mesh_colab.ipynb:
```diff
@@ -79,7 +46,9 @@
-        "!pip install imageio-ffmpeg -qqq"
+        "!pip install imageio-ffmpeg -qqq\n",
+        "!apt-get update # Added to update the package list before installing packages\n",
+        "!sudo apt -qq install imagemagick # Re-added to ensure ImageMagick is correctly installed, as it may not have been installed earlier"
```
혹시 아직 실험하시는 환경이 살아 있다면, `apt update` (`apt-get update`) 를 셀의 가장 윗쪽에서 실행하도록 하고, `apt install imagemagic` 이 그 다음에 오도록 변경해보는 것은 어떨까요?

chuchu275 replied:
그렇게 수정하도록 하겠습니다!

ProtossDragoon reviewed (commented):
고생 많으셨습니다. 레거시 버전이 잘 돌도록 환경을 맞추는 작업이 정말 열뻗치는 작업이잖아요. 어떻게든 이 코드 돌리려고 힘써 주셔서 감사하고 프로젝트 잘 마무리하셨다니 너무 축하드립니다. 코멘트 한번 확인해주시고 편히 회신 부탁드립니다. :)

@yyongjae 혹시 지난번에 COLAB 돌릴때 크레딧 남은 것 있으십니까 - 한번 돌려보려는데 내 COLAB 왜 자꾸 튕기지

chuchu275 committed:
SJChu: Code Updating for PR#30

1. colmap_colab.ipynb
- Ceres-solver 버전이 그 사이에 업데이트되어, 2.2.0 버전으로 고정해주었습니다.
- colmap의 릴리즈 태그를 고정하였습니다.

3. extract_mesh_colab.ipynb
- #apt-get update와 imagemagick을 맨 위로 올렸습니다.

ProtossDragoon reviewed (approved):
입시랑 겹치실까 걱정됐는데, 마무리까지 하시느라 고생 많으셨습니다..! 저희가 돌려보고 추가로 이슈 있으면 질문 드릴게요 :)

ProtossDragoon commented:
@all-contributors please add @chuchu275 for code.

(testing)

ProtossDragoon commented:
@all-contributors please add @chuchu275 for code.

ProtossDragoon commented:
@all-contributors please add @yunimini for code.

ProtossDragoon commented:
@all-contributors please add @Heeheegwan for code.

ProtossDragoon commented:
@all-contributors please add @namsuhan-hub for code.

ProtossDragoon commented:
@all-contributors
please add @chuchu275 for maintenance. 
please add @Heeheegwan for maintenance.
please add @yunimini for maintenance.
please add @namsuhan-hub for maintenance.
